ruby-challenges.rb

```diff
@@ -10,23 +10,113 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+=begin
+    PSEUDO CODE:
+        define a method named contains_letter
+        takes in 2 arguments: one is an array of words and two is a single letter string
+            iterate through the array of words
+            check if each word in the array includes the given letter or not
+        return an array of all the words that contain that letter
+=end
+
+def contains_letter array, letter
+    array.select{|word| word.include? letter}
+end
+
+p contains_letter beverages_array, letter_o
+p contains_letter beverages_array, letter_t
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
-# Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
+us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Vermont'] }
+# Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington']
+
+=begin
+    PSEUDO CODE:
+        define a method named alphabetical_order
+        takes in a hash with key and array values
+        access each key value pair in the hash and concatenate them
+        (is there a way to access the keys without knowing the name?)
+            sort the combined array in alphabetical order
+        returns a combined array of all the values in alphabetical order
+=end
+
+def alphabetical_order state_hash
+    p state_hash[:northwest].concat(state_hash[:southwest]).concat(state_hash[:northeast]).sort
+end
+
+alphabetical_order us_states
```

Comment:
Well done! try to think of what you could do if the keys were unknown as well...